webSocket/transceiver/transceiver.go

```diff
@@ -124,6 +138,25 @@ func (wt *wsTransceiver) verifyPayloadAndSendAckIfNeeded(connection webSocket.WS
 	wt.sendAckIfNeeded(connection, payloadData)
 }
 
+func (wt *wsTransceiver) handleAckMessage(payload []byte) {
+	counter, err := wt.payloadParser.ExtractUint64FromAckMessage(payload)
+	if err != nil {
+		wt.log.Debug("wsTransceiver.handleAckMessage cannot decode the ack message", "error", err)
+		return
+	}
+
+	wt.log.Trace("wt.handleAckMessage: received ack", "counter", counter)
+	expectedCounter := atomic.LoadUint64(&wt.counter)
+	if expectedCounter == counter {
```

Comment:
let's change the condition here in order to have sequences like if ... return, if ... return, and if all if branches pass, we can write the empty struct on ack chan

Reply:
refactored.